City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-4091: Group news paragraph

# Group news paragraph [UHF-4091](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4091)
Added a group news functionality and paragraph for the KASKO instance.

## What was done
* Enabled helfi_news_item feature and removed regular permissions from it so that it can be added ONLY to groups.
* Created paragraph with a view that lists 4 or 6 latest group news. The group is selected to the paragraph manually.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Checkout this branch.
* Update the HDBT theme
    * `composer require drupal/hdbt:dev-UHF-4091_group_news_paragraph`
* Run `make drush-cr && make drush-cim && make drush-cr`
* Run `make shell`
* Inside the container run `drush locale:check && drush locale:update && drush cr`

## How to test
* Once everything is in place go to create a group and a user that has only school editor role.
* Once you have created these things go and add the user to the group and also create front page for the group.
* Now log in with the school editor user and make sure you are able to add news for the group with the user.
* Create 7 news. You can just add the title - nothing else is needed.
* Now go to edit the front page of the group and add a paragraph for it called "Latest group news"
* Add a title and a description and select the group. It won't offer other groups than the one you are a member.
* Save and check that the paragraph looks good and works responsively.
* Layouts are here: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HEL/pages/7827357697/Uutisnostot

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/282
